cli/cmd/add/component-version/cmd.go

```diff
@@ -0,0 +1,359 @@
+	enum.Var(cmd.Flags(), FlagComponentVersionConflictPolicy, ComponentVersionOverridePolicies(), "policy to apply when a component version already exists in the repository")
+	cmd.Flags().Bool(FlagSkipReferenceDigestProcessing, false, "skip digest processing for resources and sources. Any resource referenced via access type will not have their digest updated.")
+
+	return cmd
+}
+
+func AddComponentVersion(cmd *cobra.Command, _ []string) error {
+	pluginManager := ocmctx.FromContext(cmd.Context()).PluginManager()
+	if pluginManager == nil {
+		return fmt.Errorf("could not retrieve plugin manager from context")
+	}
+
+	credentialGraph := ocmctx.FromContext(cmd.Context()).CredentialGraph()
+	if credentialGraph == nil {
+		return fmt.Errorf("could not retrieve credential graph from context")
+	}
+
+	concurrencyLimit, err := cmd.Flags().GetInt(FlagConcurrencyLimit)
+	if err != nil {
+		return fmt.Errorf("getting concurrency-limit flag failed: %w", err)
+	}
+
+	skipReferenceDigestProcessing, err := cmd.Flags().GetBool(FlagSkipReferenceDigestProcessing)
+	if err != nil {
+		return fmt.Errorf("getting skip-reference-digest-processing flag failed: %w", err)
+	}
+
+	cvConflictPolicy, err := enum.Get(cmd.Flags(), FlagComponentVersionConflictPolicy)
+	if err != nil {
+		return fmt.Errorf("getting component-version-override-policy flag failed: %w", err)
+	}
+
+	repoSpec, err := GetRepositorySpec(cmd)
+	if err != nil {
+		return fmt.Errorf("getting repository spec failed: %w", err)
+	}
+
+	cacheDir, err := cmd.Flags().GetString(FlagBlobCacheDirectory)
+	if err != nil {
+		return fmt.Errorf("getting blob cache directory flag failed: %w", err)
+	}
+
+	constructorSpec, err := GetComponentConstructor(cmd)
+	if err != nil {
+		return fmt.Errorf("getting component constructor failed: %w", err)
+	}
+
+	instance := &constructorProvider{
+		cache:          cacheDir,
+		targetRepoSpec: repoSpec,
+		pluginManager:  pluginManager,
+		graph:          credentialGraph,
+	}
+
+	opts := constructor.Options{
+		TargetRepositoryProvider:       instance,
+		ResourceRepositoryProvider:     instance,
+		SourceInputMethodProvider:      instance,
+		ResourceInputMethodProvider:    instance,
+		CredentialProvider:             instance.graph,
+		ConcurrencyLimit:               concurrencyLimit,
+		ComponentVersionConflictPolicy: ComponentVersionConflictPolicy(cvConflictPolicy).ToConstructorConflictPolicy(),
+	}
+	if !skipReferenceDigestProcessing {
+		opts.ResourceDigestProcessorProvider = instance
+	}
+
+	opts, stop, err := registerConstructorProgressTracker(cmd, opts)
+	if err != nil {
+		return fmt.Errorf("registering constructor progress tracker failed: %w", err)
+	}
+
+	_, err = constructor.ConstructDefault(cmd.Context(), constructorSpec, opts)
+	if stop != nil {
+		stop()
+	}
+
+	time.Sleep(10 * time.Millisecond)
+	return err
+}
+
+func GetRepositorySpec(cmd *cobra.Command) (runtime.Typed, error) {
+	repoRef, err := file.Get(cmd.Flags(), FlagRepositoryRef)
```

Comment:
I'm pretty sure we need a filepath.Clean here. WDYT?

Reply:
file.Get fetches a filepath. What would be the reason for a Clean here?

Reply:
The filepath is user defined isn't it? Unless I'm wrong. In that case, it's prone to injections.

Reply:
its a flag so how would that injection attack work? Im just wondering how a possible attack vector might look like. I think we can add clean in the file package of course

Reply:
Nah, you're fine, I think I misunderstood where the value is coming from.